src/lib/middleware/response-time-metrics.ts

```diff
@@ -2,21 +2,44 @@ import * as responseTime from 'response-time';
 import EventEmitter from 'events';
 import { REQUEST_TIME } from '../metric-events';
 import { IFlagResolver } from '../types/experimental';
+import { InstanceStatsService } from 'lib/services';
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
 const _responseTime = responseTime.default;
 
+const appNameReportingThreshold = 100;
+
 export function responseTimeMetrics(
     eventBus: EventEmitter,
     flagResolver: IFlagResolver,
+    instanceStatsService: Pick<InstanceStatsService, 'getStatsSnapshot'>,
 ): any {
-    return _responseTime((req, res, time) => {
-        const { statusCode } = res;
+    let appCount = appNameReportingThreshold;
 
+    return _responseTime(async (req, res, time) => {
+        const { statusCode } = res;
         const pathname = req.route ? req.baseUrl + req.route.path : '(hidden)';
 
+        // async get of appCount (only if promise is settled)
+        instanceStatsService
+            .getStatsSnapshot()
+            .then((stats) => {
+                const currentCount = stats?.clientApps?.find(
+                    (t) => t.range === '7d',
+                )?.count;
+                // update app count
+                appCount =
+                    currentCount === undefined
+                        ? appNameReportingThreshold
+                        : currentCount;
+            })
+            .catch(() => {});
```

Comment:
I'm not sure if this can be done better. It basically inspects the promise, and if it's resolved it uses the stats information to set the `appCount` variable. If not settled, it just moves on. 

This avoids increasing latency on responseTime calculation